libs/sdk-core/src/breez_services.rs

```diff
@@ -2052,33 +2051,30 @@ impl Receiver for PaymentReceiver {
                 // remove the fees from the amount to get the small amount on the current node invoice.
                 destination_invoice_amount_msat = req.amount_msat - channel_fees_msat;
             }
+
+            let lsp_hop = RouteHint {
+                hops: vec![RouteHintHop {
+                    src_node_id: lsp_info.pubkey,
+                    short_channel_id: parse_short_channel_id("1x0x0")?,
+                    fees_base_msat: lsp_info.base_fee_msat as u32,
+                    fees_proportional_millionths: (lsp_info.fee_rate * 1000000.0) as u32,
+                    cltv_expiry_delta: lsp_info.time_lock_delta as u64,
+                    htlc_minimum_msat: Some(lsp_info.min_htlc_msat as u64),
+                    htlc_maximum_msat: None,
+                }],
+            };
+            info!("Adding LSP hop as routing hint: {:?}", lsp_hop);
+            routing_hints.push(lsp_hop)
         } else {
-            // not opening a channel so we need to get the real channel id into the routing hints
+            // not opening a channel so we need to get all private channels into the routing hints
             info!("Finding channel ID for routing hint");
-            for peer in self.node_api.list_peers().await? {
-                if hex::encode(peer.id) == lsp_info.pubkey && !peer.channels.is_empty() {
-                    let active_channel = peer
-                        .channels
-                        .iter()
-                        .find(|&c| c.state == ChannelState::Opened)
-                        .ok_or_else(|| anyhow!("No open channel found"))?;
-                    let hint = active_channel
-                        .alias_remote
-                        .clone()
-                        .or(active_channel.short_channel_id.clone());
-
-                    short_channel_id = match hint {
-                        None => None,
-                        Some(hint) => {
-                            let scid = parse_short_channel_id(&hint)?;
-                            info!("Found channel ID: {scid:?} {active_channel:?}");
-                            Some(scid)
-                        }
-                    };
+            routing_hints = self.node_api.get_routing_hints().await?;
```

Comment:
The log statement above is not relevant anymore, as this is not finding the channel ID anymore.

Reply:
fixed